### PR TITLE
Title page - Videos displayed higher

### PR DESCRIPTION
**Pull request by KvnMrs**
Issue: https://github.com/blockframes/blockframes/issues/6893

### CODE REVIEW

File: libs/movie/src/lib/movie/marketplace/main/main.component.scss [hunk 1]
```diff
@@ -32,8 +32,19 @@ h5 {
   }
 }
 
+.mains-information{
+  display: flex;
+}
+
+.video {
+  padding-right: 32px;
+}
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/marketplace/main/main.component.scss:41]:
```suggestion
.mains-information{
  display: flex;
  .video {
    padding-right: 32px;
  }
}
```

File: libs/movie/src/lib/movie/marketplace/main/main.component.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Component, ChangeDetectionStrategy, OnInit } from '@angular/core';
 import { TitleMarketplaceShellComponent } from '../shell/shell.component';
-import { Movie } from '@blockframes/model';
+import { App, Movie } from '@blockframes/model';
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/marketplace/main/main.component.ts:3]: app is not used

File: libs/movie/src/lib/movie/marketplace/main/main.component.ts [hunk 3]
```diff
@@ -32,16 +32,18 @@ export class MainComponent implements OnInit {
...
       this.alreadyPlayed = true;
     }
   }
+
+  showOneVideo(movie: Movie) {
+    const video = movie.promotional.videos;
+    if (video.salesPitch?.storagePath) return video.salesPitch;
+    if (video.otherVideos[0]?.storagePath) return video.otherVideos[0];
+    if (video.publicScreener?.storagePath) return video.publicScreener;
+    if (video.screener?.storagePath) return video.screener;
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/marketplace/main/main.component.ts:47]: I think you should look for `.jwPlayerId`

Comment (phpgeekfr) [reply libs/movie/src/lib/movie/marketplace/main/main.component.ts:47]: and also look for `privacy === 'public'`

Comment (KvnMrs) [reply libs/movie/src/lib/movie/marketplace/main/main.component.ts]: i already look `.jwPlayerId` in the template, if i add another condition for check it in the `.ts` i get an infinite load video

File: libs/movie/src/lib/movie/marketplace/main/main.component.ts [hunk 4]
```diff
@@ -32,16 +32,20 @@ export class MainComponent implements OnInit {
...
       this.alreadyPlayed = true;
     }
   }
+
+  showOneVideo(movie: Movie) {
+    const video = movie.promotional.videos;
+    if (video.salesPitch?.storagePath && video.salesPitch?.privacy === 'public') return video.salesPitch;
+    if (video.otherVideos[0]?.storagePath && video.otherVideos[0]?.privacy === 'public') return video.otherVideos[0];
+    if (video.publicScreener?.storagePath && video.publicScreener?.privacy === 'public') return video.publicScreener;
+    if (video.screener?.storagePath && video.screener?.privacy === 'public') return video.screener;
```
[the remark this answers is not in the document]
Comment (RemcoSimonides) [reply libs/movie/src/lib/movie/marketplace/main/main.component.ts:47]: Correct. A screener cannot be public. A public screener is always public.

[the remark this answers is not in the document]
Comment (cams-goussale) [reply libs/movie/src/lib/movie/marketplace/main/main.component.ts:47]: @phpgeekfr I think it's fine if the video is repeated

File: libs/movie/src/lib/movie/marketplace/main/main.component.ts [hunk 5]
```diff
@@ -32,16 +32,20 @@ export class MainComponent implements OnInit {
...
   }
+
+  getMainVideo(movie: Movie) {
+    const video = movie.promotional.videos;
+    const otherVideo = video.otherVideos.filter(video => video.privacy === 'public' && video.storagePath !== '' && video.jwPlayerId !== '')
+    if (otherVideo.length) return otherVideo[0];
+    if (video.salesPitch?.storagePath && video.salesPitch?.privacy === 'public' && video.salesPitch?.jwPlayerId) return video.salesPitch;
+    if (video.publicScreener?.storagePath && video.publicScreener?.privacy === 'public' && video.publicScreener?.jwPlayerId) return video.publicScreener;
+    if (video.screener?.storagePath && video.screener?.jwPlayerId) return video.screener;
+  }
```
Comment (phpgeekfr) [inline libs/movie/src/lib/movie/marketplace/main/main.component.ts:49]: Sales Pitch should be first as mentioned in the issue.
publicScreener is always public, no need to check but can be displayed on catalog app only
screener cannot be public, no need here.

```suggestion
  getMainVideo(movie: Movie) {
    const videos = movie.promotional.videos;
    const isPublicVideo = (video: MovieVideo) => {
      return video?.storagePath && video?.privacy === 'public' && video?.jwPlayerId;
    }

    if (isPublicVideo(videos.salesPitch)) return videos.salesPitch;
    const otherVideo = videos.otherVideos.find(video => isPublicVideo(video));
    if (otherVideo) return otherVideo;
    if (this.shell.currentApp === 'catalog' && isPublicVideo(videos.publicScreener)) return videos.publicScreener;
  }
```

**Review by phpgeekfr — APPROVED** [review]

**Pull request merged**